src/index.js

```diff
@@ -1,54 +1,127 @@
 import './style.css';
 import reIcon from './recycle.png';
 import tdIcon from './3dotd.png';
+import adIcon from './enter.png';
 
+class ToDoList {
+  constructor() {
+    this.tasks = JSON.parse(localStorage.getItem('todolist')) || [];
+    this.ulList = document.getElementById('to-do-list');
+    this.addListInput = document.createElement('input');
+    this.addListButton = document.createElement('button');
+  }
+
+  initialize() {
+    this.renderTasks();
+    this.setupEventListeners();
+    this.setupAddTaskInput();
+  }
+
+  renderTasks() {
+    this.ulList.innerHTML = ''; // Clear the previous list before rendering
+    this.tasks.forEach((task) => {
+      const taskItem = document.createElement('li');
+      taskItem.className = 'each-task';
+
+      const checkTag = document.createElement('div');
+      checkTag.className = 'check-tag';
+
+      const checkbox = document.createElement('input');
+      checkbox.type = 'checkbox';
+      checkbox.className = 'task-check';
+      checkbox.checked = task.complete;
+
+      const taskDesc = document.createElement('input');
+      taskDesc.type = 'text';
+      taskDesc.className = 'task-desc';
+      taskDesc.value = task.description;
+
+      const deleteButton = document.createElement('button');
+      deleteButton.className = 'delbin';
+      deleteButton.type = 'button';
+
+      const deleteButtonImg = new Image();
+      deleteButtonImg.src = tdIcon;
+      deleteButtonImg.alt = 'threedots';
+      deleteButtonImg.className = 'three-dots';
+
+      checkTag.appendChild(checkbox);
+      checkTag.appendChild(taskDesc);
+      taskItem.appendChild(checkTag);
+      deleteButton.appendChild(deleteButtonImg);
+      taskItem.appendChild(deleteButton);
+
+      checkbox.addEventListener('change', () => {
+        task.complete = checkbox.checked;
+        localStorage.setItem('todolist', JSON.stringify(this.tasks));
+        taskItem.classList.toggle('checked');
+      });
+
+      taskDesc.addEventListener('input', () => {
+        task.description = taskDesc.value;
+        localStorage.setItem('todolist', JSON.stringify(this.tasks));
+      });
+
+      deleteButton.addEventListener('click', () => {
+        this.tasks = this.tasks.filter((t) => t.id !== task.id);
+        localStorage.setItem('todolist', JSON.stringify(this.tasks));
+        this.renderTasks();
+      });
+
+      if (task.complete) {
+        taskItem.classList.add('checked');
+      }
+
+      this.ulList.appendChild(taskItem);
+    });
+  }
+
+  addTask(description) {
+    const newTask = {
+      id: this.tasks.length + 1,
+      description,
+      complete: false,
+    };
+    this.tasks.push(newTask);
+    localStorage.setItem('todolist', JSON.stringify(this.tasks));
+    this.renderTasks();
+  }
+
```

Comment:
- [ ] Dear @mj-e-boc, As stated in the project requirement, we need to create a new file for the new functionalities, this includes ` A function for adding a new task`, `A function for editing task descriptions` and `A function for deleting a task`. Please implement these functionalities in a new file. 
- [ ] Also, While adding a new task, we need to use a property `index` instead of `Id` besides that deleting a task should update all remaining items' indexes, so they represent the current list order and are unique. currently your `id` is not getting updated once I remove an element from the list. Please take a look and made some changes. 

### Tasks list after deleting the second task 
![image](https://github.com/mj-e-boc/To-Do-List/assets/69077061/d09dcf30-a222-4828-b410-982466667be8)